pkgs/applications/networking/mailreaders/notmuch/default.nix

```diff
@@ -93,6 +93,16 @@ stdenv.mkDerivation rec {
     moveToOutput bin/notmuch-emacs-mua $emacs
   '' + optionalString withVim ''
     make -C vim DESTDIR="$out/share/vim" prefix="" install
+  '' + optionalString (!isNull ruby) ''
+    make -C bindings/ruby install DESTDIR="" prefix="$out" install
+    # This is very specific, see pkgs/development/interpreters/ruby/default.nix
+    mkdir -p $out/nix-support
+    cat > $out/nix-support/setup-hook <<EOF
```

Comment:
Could you explain a bit more how does this work? Perhaps you can alternatively patch `notmuch.vim` to use a ruby interpreter that has the bindings from `bindings/ruby`.